enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,940 @@
+  - "@dhellmann"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2023-04-14
+last-updated: 2023-05-09
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/USHIFT-518
+see-also:
+  - "/enhancements/microshift/microshift-greenboot.md"
+  - "/enhancements/microshift/etcd-supportability.md"
+replaces:
+  - https://github.com/openshift/enhancements/pull/1312
+superseded-by:
+  - None
+---
+
+# MicroShift updateability in ostree based systems
+
+## Summary
+
+This enhancement focuses on high level overview of updating
+MicroShift running on ostree based systems such as RHEL 4 Edge.
+Enhancement covers backup and restore of MicroShift data,
+version migration (upgrade and downgrade) of MicroShift and
+its consequences (migration of data between schema versions),
+and interactions with GreenBoot and operating system.
+
+## Motivation
+
+MicroShift team is working towards a general availability (GA) release.
+As GA product, it is expected it can be updated to
+apply security patches and bug fixes, and leverage features
+in newer version while keeping and using existing data.
+
+MicroShift is intended to be a part of Red Hat Device Edge (RHDE) which is based
+on RHEL For Edge (R4E) which is immutable Linux distribution by leveraging
+[ostree](https://ostreedev.github.io/ostree/) technology.
+It allows changing root filesystem for next boot by staging new commits or
+rolling back to previous one.
+ostree is commonly paired with [greenboot](https://github.com/fedora-iot/greenboot)
+which provides automated health assessment of the system and trigger a rollback
+if rebooting the device doesn't result in device becoming healthy.
+
+Even though, OpenShift does not support downgrade or rollback, MicroShift must
+support it in some form to fit into RHDE.
+Rollback (going back to older deployment) will be supported only if MicroShift
+ran on that deployment and data compatible with that deployment was backed up.
+Downgrade (migrating to older version of MicroShift) will not be supported.
+
+In order to integrate into RHDE, MicroShift needs to be augmented with
+functionality to back up and restore its data together with ostree deployments,
+and refuse or perform data migration to newer storage schema version.
+Integration with greenboot will allow system's health to depend on state of the
+MicroShift and will provide necessary information to manage backups.
+
+### User Stories
+
+* As a MicroShift administrator, I want to safely update MicroShift
+  so that I can get bug fixes, new features, and security patches.
+* As a MicroShift administrator, I want my system to roll back
+  to a previous good configuration when greenboot fails.
```

Comment:
Added, please check if that's what you meant

Reply:
> I want it to roll back to a previous healthy state expecting MicroShift to be
  just like before the failed upgrade.
  
  i guess that implies "restoring etcd runtime data/cluster state to what it was before the upgrade attempt" but it could be more explicit.